tests/server/test_compilerservice.py

```diff
@@ -795,6 +800,9 @@ async def run_compile_and_wait_until_compile_is_done(
     # prevent race conditions where compile request is not yet in queue
     await retry_limited(lambda: not compiler_queue.empty(), timeout=10)
     run = compiler_queue.get(block=True)
+    if fail is not None:
```

Comment:
Is it intended that we can make it fail by passing it "True" or "False"?

Reply:
no, if it is set, it will use that value. 

1. if `fail_on_pull` is None, it will not set the value, using the default set when constructing the mock
2. if `fail_on_pull` is True, it will set it to fail
2. if `fail_on_pull` is False, it will set it to succed

Reply:
woof, I think I convinced myself that something was wrong, sorry for that.